tests/unit/test_job_file.py

```diff
@@ -60,13 +82,18 @@ def test_write_prelude_invalid_cylc_command(mocked_glbl_cfg):
 @mock.patch.dict(
     "os.environ", {'CYLC_SUITE_DEF_PATH': 'cylc/suite/def/path'})
 @mock.patch("cylc.flow.job_file.get_remote_suite_run_dir")
-def test_write(mocked_get_remote_suite_run_dir):
+def test_write(mocked_get_remote_suite_run_dir, fixture_get_platform):
     """Test write function outputs jobscript file correctly"""
     with NamedTemporaryFile() as local_job_file_path:
-        local_job_file_path = local_job_file_path.name
+        local_job_file_path = str(local_job_file_path)
```

Comment:
```suggestion
        local_job_file_path = local_job_file_path.name
```
This test is currently producing temporary files that are not cleaned up after the test has finished. This fixes the problem (credit to @kinow for the original fix)